chain/chain/src/resharding.rs

```diff
@@ -0,0 +1,182 @@
+/// Implementation for all resharding logic.
+/// StateSplitRequest and StateSplitResponse are exchanged across the client_actor and sync_jobs_actor.
+/// build_state_for_split_shards_preprocessing and build_state_for_split_shards_postprocessing are handled
+/// by the client_actor while the heavy resharding build_state_for_split_shards is done by sync_jobs_actor
+/// so as to not affect client.
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use near_chain_primitives::error::Error;
+use near_client_primitives::types::StateSplitApplyingStatus;
+use near_o11y::log_assert;
+use near_primitives::hash::CryptoHash;
+use near_primitives::shard_layout::{account_id_to_shard_uid, ShardLayout};
+use near_primitives::state_part::PartId;
+use near_primitives::syncing::{get_num_state_parts, STATE_PART_MEMORY_LIMIT};
+use near_primitives::types::chunk_extra::ChunkExtra;
+use near_primitives::types::{AccountId, ShardId, StateRoot};
+use near_store::split_state::get_delayed_receipts;
+use near_store::{ShardTries, ShardUId, Trie};
+
+use tracing::debug;
+
+use crate::types::RuntimeAdapter;
+use crate::Chain;
+
+#[derive(actix::Message)]
+#[rtype(result = "()")]
+pub struct StateSplitRequest {
+    pub runtime_adapter: Arc<dyn RuntimeAdapter>,
+    pub sync_hash: CryptoHash,
+    pub shard_id: ShardId,
+    pub shard_uid: ShardUId,
```

Comment:
seems a bit redundant (shard_uid contains shard_id) it's something like before and after resharding
anyway let's not change anything in this PR and keep it as pure refactor but if you have a moment you can see what's up with this

Reply:
Yeah, I honestly don't know why is it done this way. Probably an oversight in the original implementation.